modules/module_service.go

```diff
@@ -104,19 +103,14 @@ func (w *moduleService) run(serviceContext context.Context) error {
 
 func (w *moduleService) stop(_ error) error {
 	var err error
-	switch w.service.State() {
-	case services.Running:
+	if w.service.State() == services.Running {
```

Comment:
This reverts to previous logic, as underlying service can't be in `Starting` state when we get here.